pkg/feeder/feeder.go

```diff
@@ -52,15 +54,18 @@ func NewClient(baseURL, baseAPI string, client *HttpClient) *Client {
 	}
 
 	// retry mechanism for do requests
+	// https://cloud.google.com/iot/docs/how-tos/exponential-backoff
 	retryFuncForDoReq := func(req *http.Request, httpClient HttpClient) (*http.Response, error) {
 		var res *http.Response
-		wait := 2 * time.Second
-		for {
+		wait := 1*time.Second + time.Duration(rand.Intn(1000))*time.Millisecond
+		maxWait := 32 * time.Second
+		for i := 1; i <= 10; i++ {
 			res, err = httpClient.Do(req)
 			if err != nil || res == nil || res.StatusCode != http.StatusOK {
-				wait *= 2
 				Logger.With("Waiting:", wait.Seconds(), "Error", err).Info("Waiting to do again a request")
 				time.Sleep(wait)
+				x := time.Duration(math.Pow(2, float64(i)))*time.Second + time.Duration(rand.Intn(1000))*time.Millisecond
+				wait = minDuration(x, maxWait)
```

Comment:
You are slightly complicating this.

- The line `wait *= 2` that you just deleted describes `2^n` and it doesn't require maintaining an iteration variable or calling `math.Pow`.
- You do not need to define your own minimum function. Use `math.Min` with a cast. The values we are dealing with are guaranteed to stay within the appropriate range.

Reply:
Changing `math.Pow` to `wait *= 2` does slightly change the logic, as we are now also doubling the random part every time, but this shouldn't be a problem for our use case.

Reply:
Remove the random element. It adds unnecessary complexity.